packages/nimble-components/src/table/specs/table-row-grouping-hld.md

```diff
@@ -0,0 +1,228 @@
+```
+
+### Rendering group header values
+
+Rendering group header values have similar concerns as rendering cell values, providing reason to adopt the pattern outlined in [this PR](https://github.com/ni/nimble/pull/1052) for columns to specify a custom element to use to render the value. As shown in the mixin snippet above, it will provide an abstract `groupHeaderViewTag` property, which will require groupable columns to specify the element to use to display the header value. This element will derive from an element of type `TableGroupRowHeaderView`:
+
+```
+export interface TableGroupRowHeaderValueState<TColumnConfig = unknown> {
+    groupHeaderValue: TableFieldValue;
+    columnConfig: TColumnConfig;
+}
+
+export abstract class TableGroupHeaderView<TColumnConfig> implements TableGroupRowHeaderValueState<TColumnConfig> {
+    @observable
+    groupHeaderValue: TableFieldValue;
+
+    @observable
+    columnConfig: TColumnConfig;
+}
+```
+
+The value of the `groupHeaderValue` property will be sourced from the TanStack [`groupingValue` property](https://tanstack.com/table/v8/docs/api/features/grouping#groupingvalue) of a given TanStack `Row` instance. So, this will be a raw, unformatted value. Any formatting of the value for display purposes is up to the concrete implementation of a particular `TableGroupRowHeaderView`. The `columnConfig` value comes from the `TableColumn` instance used for grouping values by.
+
+A naive implementation for the custom element to use for rendering group header values for the `TableColumnText` might look like the following:
+
+```
+export class TableColumnTextGroupHeaderView extends TableGroupHeaderView<
+TableColumnTextColumnConfig
+> {
+    @observable
+    public override columnConfig!: TableColumnTextColumnConfig;
+
+    @volatile
+    public get content(): string {
+        return typeof this.groupHeaderValue === 'string'
+            ? this.groupHeaderValue
+            : this.columnConfig.placeholder;
+    }
+}
+
+const tableColumnTextGroupHeaderView = TableColumnTextGroupHeaderView.compose({
+    baseName: 'table-column-text-group-header',
+    template: html<TableColumnTextGroupHeaderView>`
+        <nimble-text-field readonly="true" value="${x => x.groupHeaderValue}" placeholder="${x => x.columnConfig.placeholder}">
+        </nimble-text-field>`,
+    styles: /* styling */
+});
+DesignSystem.getOrCreate().withPrefix('nimble').register(tableColumnTextGroupHeaderView());
+```
+
+### Rendering the row group
+
+Rendering a row group has concerns beyond just rendering the grouping value. It also has a button for controlling the expand/collapse state of a row group (indented for sub-groups) as well as potentially rendering auxiliary information such as the total number of items within a group. We can provide a new `TableGroupRow` element to manage all of these:
+
+```
+export interface TableGroupRowState {
+    rowGroupValue: unknown;
+    level: number;
+    leafItemCount: number;
+    columnConfig: unknown;
```

Comment:
Maybe? Ultimately, that would be a generic constraint that will never be given a concrete type. Assuming this interface goes away, and this API gets separated out into individual properties on `TableGroupRow`, I'm not sure what value we have in genericising `TableGroupRow`. There's even an argument that nothing is gained by having `TableRow` being generic on `TDataRecord` as it is now, instead of just declaring its `dataRecord` property of type `TableRecord`.

Reply:
I will consider this suggestion in the actual implementation.